webapp/IronCalc/src/components/FormatMenu/FormatMenu.tsx

```diff
@@ -26,6 +33,12 @@ const FormatMenu = (properties: FormatMenuProps) => {
     [properties.onChange],
   );
 
+  const isCustomFormat = useMemo(() => {
+    return !Object.values(NumberFormats).includes(
+      properties.numFmt as NumberFormats,
+    );
+  }, [properties.numFmt]);
+
```

Comment:
useMemo is too much here. It is used when you have computationally expensive operations you want to memoize.

This technique is used when you want to store values of functions. Say `f` is a computationally expensive function to compute. Imagine the someone asks for `f(4)` you go ahead and compute it. It turns out it is `42`. Next time someone asks for `f(4)` you just return 42, no need to compute it again.

Perhaps a better way. In `formatUtil.ts` add at the end:

`export const KNOWN_FORMATS = new Set<string>(Object.values(NumberFormats));`

Then in here (after importing KNOWN_FORMATS)

`const isCustomFormat = !KNOWN_FORMATS.has(properties.numFmt);`

Reply:
Thanks!